### ISSUE TITLE
Years of Frustration with Subscription Creator

### ISSUE THREAD

**Issue by LuvBiglyBT** [BiglySoftware/BiglyBT#826]
After many years of getting error messages when I try to create a subscription within Vuze/BiglyBT, I am asking for help from the community.  In years past, I would have submitted it to the community forum but this seems to be the only forum available for anything, so I apologize in advance.

Recently, LimeTorrents RSS generator has been non-functional.  Without working search templates for BiglyBT it is impossible to easily create subscriptions.  At this point I am back to weighing through individual searches on torrent sites.  It really is slow and wasted time when it used to be so easy.

Over the years I must have tried at least 30 times to create a subscription from scratch and none of them have produced any results in preview, so I am asking Parg or anyone else to just lay it out in step by step terms how to create a subscription 9a search template for LimeTorrents would also be appreciated.

I have never claimed to have the coding knowledge to know how to do this and the warning that to create a subscription needs advanced knowledge did not daunt me from trying again and again.

Maybe we could incorporate[orate the RSS generator that LimeTorrents used to use before it stopped working?  Or maybe we could make the subscription process easier to follow?  Of course search template that are functional would have to be included, although I understand why they were eliminated years ago.

Simply put, I am just trying to create a subscription or RSS feed.  Any of you brainiacs care to help out a coding dummy? :)

**Comment by parg** [comment]
I can't help with any specific torrent sites unfortunately although I feel your pain at dealing with the search template creator. Note that some sites refuse to play nicely with any kind of automated access (e.g. the employ Cloud Front to help with (D)DOS protection) and you'll probably never get anything working.

**Comment by Havokdan** [comment]
How about an easier way to use Jackett through BiglyBT?

**Comment by parg** [comment]
Feel free to investigate how this could be made easier and report back

**Comment by parg** [comment]
OK, I installed it - created a feed, pasted it into BiglyBT address bar, it created a subscription and it all works - what's the problem?

**Comment by Havokdan** [comment]
I did not say it did not work, just that it was complicated for me. Mainly that Jackett does search on multiple site at the same time, it should be possible to show through BiglyBT.

**Comment by parg** [comment]
"it should be possible" - how? I can't work on everything in the world myself you know

**Comment by parg** [comment]
So the manual Jackett search gives results at

http://127.0.0.1:9117/api/v2.0/indexers/all/results?apikey=<your jackett api key>&Query=bunny&_=1548156611274

as JSON - create a JSON based search template in BiglyBT based on this

**Comment by parg** [comment]
Here's one:

{"vuze":{"components":[{"content":{"web.time_zone":"GMT-11","web.required_cookies":[],"name":"Jackett","select_rec":1,"uid":"\\xd88b5ff4189272385eaf\\x","last_updated":1548157139579,"az_version":5,"version":2,"selected":2,"source":2,"id":2929201020,"l1_map":[{"from":6,"to":6,"maps":[]}],"l2_map":[{"from":6,"to":8,"maps":[]}],"web.search_url_format":"http:\/\/127.0.0.1:9117\/api\/v2.0\/indexers\/all\/results?apikey=*****************************************&Query=%s&_=1548156611274","web.needs_auth":0,"web.auto_date":1,"json.path":"Results","rank_bias":"1.0","web.maps":[{"field":6,"name":"CategoryDesc"},{"field":1,"name":"Title"},{"field":102,"name":"Link"},{"field":103,"name":"Comments"},{"field":2,"name":"PublishDate"},{"field":3,"name":"Size"},{"field":5,"name":"Seeders"},{"field":4,"name":"Peers"},{"field":200,"name":"InfoHash"}],"pref_count":"0.0","web.full_cookies":null,"type":2,"web.auth_method":"transparent"},"type":1}]}}

You HAVE to replace *************************** in the above with your Jackett API key (top right in the Jacket UI). Then import this by Right--clicking on the search icon in BiglyBT (left of the address bar) and select Templates->Import JSON from clipboard

**Comment by parg** [comment]
meh, just came across a bug that limits JSON search template results to 9 :( 1801_B01 will fix that

**Comment by Havokdan** [comment]
Bugs Life, dammed, wainting next stable release, thanks parg

**Comment by LuvBiglyBT** [comment]
Ok gentlemen, I had no idea that I would spark the debate that it has, but I am glad that Havokdan saw fit to respond.  Over the years with Vuze and BiglyBT  he has offered much sound advice.  Thanks Havokdan for the suggestion about Jackett!  After muddling through creating an RSS feed with my specific parameter needs, I eventually was able to add what seems to be a functional subscription.  I say "seems" because I am still awaiting the search site to see a new entry to download, but I have high hopes.

Parg, I am aware of the limitations with direct inclusion or approval by BiglyBT of specific search templates.  I was just commenting on how much easier it was years ago when they were included.  I did not expect them to all of a sudden appear in the next update.  In fact, I would be surprised if they ever appear in a future update.  I know you cannot fix everything in the program and if I had the coding skills, I would offer my talents to the project.  Instead I contribute via donation.

I have stated before that all the continued work on the tags portion of the program does not mean much to me, probably because I do not understand their purpose or how it will help me with facilitating my use of the program.  Without a community help blog, it is difficult to get info on tags and I am reticent to be wasting your time by bringing it up in this beta issue forum.  

My point is that search templates and subscription functions have essentially been unavailable to the average user and BiglyBt MIGHT benefit by somehow simplifying the process.  If that remains low on your priority list, I am more than accepting of that.  For now, Havkdan opened the way to an alternative third party way of producing a subscription and adding it to BiglyBT, although creating a search template from Jackett is still confounding my rudimentary coding skills :).  Thanks again, Havokdan!  And thanks Parg for all the work and effort that you continue to put into BiglyBT!

**Comment by parg** [comment]
Thanks! You should be able to get the Jackett search template working fairly easily (famous last words) by following the above instructions (or check https://github.com/BiglySoftware/BiglyBT/wiki/Search-Templates for the same basic info). Tell me if you still can't get it to work and I'll try and help!

**Comment by LuvBiglyBT** [comment]
Parg,

As stated in my last comment, I have been able to create specific subscriptions, however, search templates are still eluding me.  When I try to go into BiglyBT’s search template I do not even know if what I try to paste in should be under json or regular expression.  I have tried both (probably pasting the wrong things in) with no production of results.

I really am not of the mind to bother you with such basic things created by my coding ignorance.  

Eventually, I will write up step by step directions to add a subscription for dummies, using Jackett

Thanks again for all your continued work!

LuvBiglyBT

Sent from Mail for Windows 10

**Comment by parg** [comment]
You don't need to go into the template editor to add the text that you have edited to include your API key ( the "{"vuze":{"components":[{"c...." stuff). Just copy the modified text to your clipboard and then go here by right-clicking on the little search icon:

![image](https://user-images.githubusercontent.com/395293/51553145-0cb3fe00-1e6a-11e9-84f7-befc62f6fefb.png)

**Comment by LuvBiglyBT** [comment]
Holy Cow!!!  I have a functioning search template for the first time in YEARS!  I see the issue with only 9 results and look forward to B1.

Just to let you know, I first tried to copy the RSS feed directly from Jackett and it did nothing.  Then I edited your suggested json and VOILA!

As always, you have gone beyond the norm for an administrator.  I will submit “Subscriptions for Dummies” ASAP.

Thanks Parg!

Sent from Mail for Windows 10

**Comment by Havokdan** [comment]
> Thanks! You should be able to get the Jackett search template working fairly easily (famous last words) by following the above instructions (or check https://github.com/BiglySoftware/BiglyBT/wiki/Search-Templates for the same basic info). Tell me if you still can't get it to work and I'll try and help!

Works in stable?

**Comment by parg** [comment]
To get more than 9 results you need to join the beta program and update to 1801_B01 - it is now available. You can always leave the beta program at that point if you don't want further updates.

**Comment by LuvBiglyBT** [comment]
Been in the beta program for years!  Thanks, although, for now I cannot even get the program to recognize that I do not have 1801.  It probably is just a matter of time.

BTW, needless to say, with a functional search engine, people will not have to go through the gyrations of directly adding a subscription via Jackett so my offer to write up “Subscriptions for Dummies” may be mute.  Let me know if it would be of value or whether I should just forget it.

Love that new search template wiki!

Sent from Mail for Windows 10

**Comment by Havokdan** [comment]
I do not know what I did wrong, but I could not make it work, I'll keep trying here, thanks.

**Comment by LuvBiglyBT** [comment]
LOL, Havokdan…if I can do it, ANYONE can do it 😊.  Copy Parg’s json, paste it into Wordpad or other text editor, get rid of the ………………… and replace it with your apikey assigned by Jackett, save it and copy it.  Right click on the search icon in bigly, choose Templates and then import json from clipboard and you should get a message box asking if you want to add a new template. Say yes and it should be done.

Sent from Mail for Windows 10

**Comment by Havokdan** [comment]
It still has not worked, I should have left something that should not be marked, I tried to upgrade to beta, but it says it's updated.

**Comment by parg** [comment]
B01 is available now, sorry, messed it up earlier.

**Comment by LuvBiglyBT** [comment]
B! is downloaded and working great!  Just a note…if you are indexing more than one site on Jacket, you get results for them all.  That is understandable since the json does not seem to provide specificity.  Easy fix is to use only one indexer to limit redundant results.

Havokdan, first, sometimes I have had to press enter twice occasionally after entering a search term.  Second, have you checked if any of your security on your computer is blocking the Jackett site?  I use PeerBlock and find that it is often the culprit (It is only doing its job).

Sent from Mail for Windows 10

**Comment by LuvBiglyBT** [comment]
Sorry that I cannot help you more.  Not only is mine working, but following the B1 update, I am getting many results.  The only other thing that I can suggest is to take down ALL your security just long enough to see if it makes a difference.  I had to do that to figure out why I could not get any updates to the program.  I widdled it down to a specific web server that was being blocked by PeerBlock, so I have to disable Peerblock if I want to get updates.

BTW Havokdan, this is LuvVuze from the Vuze era.  We have had many great interactions in the past and it was you who turned me on to BiglyBT.  Thanks for all your assistance, including the suggestion to look into Jackett!!

Sent from Mail for Windows 10

**Comment by Havokdan** [comment]
Just to consign, I managed to make it work, I just do not know what caused the problem, I did everything right, however BiglyBT, perhaps for some application or option that I use in Windows 10, did not want to copy the model with api key, text always ran with asterisks in place of api key, I went to edit the template and corrected and immediately worked. Thank you.

**Comment by LuvBiglyBT** [comment]
I am  pleased to hear that you met with success Havokdan!
